src/main/java/com/faforever/client/chat/PircBotXChatService.java

```diff
@@ -141,33 +142,15 @@
    */
   private boolean autoChannelsJoined;
 
-  @Inject
-  public PircBotXChatService(PreferencesService preferencesService, UserService userService, TaskService taskService,
-                             FafService fafService, I18n i18n, PircBotXFactory pircBotXFactory,
-                             @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection") ThreadPoolExecutor threadPoolExecutor,
-                             EventBus eventBus, ClientProperties clientProperties, PlayerService playerService) {
-    this.preferencesService = preferencesService;
-    this.userService = userService;
-    this.taskService = taskService;
-    this.fafService = fafService;
-    this.i18n = i18n;
-    this.pircBotXFactory = pircBotXFactory;
-    this.threadPoolExecutor = threadPoolExecutor;
-    this.eventBus = eventBus;
-    this.clientProperties = clientProperties;
-    this.playerService = playerService;
-
+  @Override
+  public void afterPropertiesSet() {
+    eventBus.register(this);
     connectionState = new SimpleObjectProperty<>(ConnectionState.DISCONNECTED);
     eventListeners = new ConcurrentHashMap<>();
     channels = observableHashMap();
     chatChannelUsersByChannelAndName = observableMap(new TreeMap<>(String.CASE_INSENSITIVE_ORDER));
     unreadMessagesCount = new SimpleIntegerProperty();
     identifiedFuture = new CompletableFuture<>();
-  }
-
-  @Override
-  public void afterPropertiesSet() {
-    eventBus.register(this);
     fafService.addOnMessageListener(SocialMessage.class, this::onSocialMessage);
```

Comment:
I think u need to add constructor back here much of this is initialization

Reply:
i moved initialization to declaration. the rest that is left in afterPropertiesSet was in PostConstruct before I changed the file, so I think the rest might stay here.